packages/react-day-picker/src/components/Day/hooks/useDay.tsx

```diff
@@ -146,6 +144,7 @@ export function useDay(
       range.onDayClick?.(date, modifiers, e);
     }
     context.onDayClick?.(date, modifiers, e);
+    changeInitialFocus(date);
```

Comment:
Wouldn’t the day already be focused by this click event? Or is it that our internal focus state doesn’t know?

Reply:
When the user clicks on the button I think the focus event will trigger and this code will set the focus day?

https://github.com/gpbl/react-day-picker/blob/9c5789c78d5cc447be2b5a77af3b406ab8fc5099/packages/react-day-picker/src/components/Day/hooks/useDay.tsx#L152